pandas/_libs/parsers.pyx

```diff
@@ -989,6 +981,10 @@ cdef class TextReader:
                         col_dtype = self.dtype[name]
                     elif i in self.dtype:
                         col_dtype = self.dtype[i]
+                    else:
```

Comment:
umm, can we not leave the original and simply copy self.dtype?

Reply:
Actually I had done that originally (send a copy of dtype, so that the original is unaffected). But then you suggested to try and not modify the dict at all(https://github.com/pandas-dev/pandas/pull/42508#discussion_r670026684) :sweat_smile: . Also @phofl had mentioned that this was a unwanted side effect of https://github.com/pandas-dev/pandas/pull/41411. So, I tried to incorporate those comments.

Reply:
please let me know if you want me to revert all the changes, and just pass a copy of dtype dictionary

Reply:
I would prefer keeping the mangle_dup_cols logic together, e.g. simply copying the dict would be the best here. If we spread this over the code base it will be harder to refactor